ethers-core/src/types/crypto/hash.rs

```diff
@@ -0,0 +1,85 @@
+//! This is a helper module used to pass the pre-hashed message for signing to the
+//! `sign_digest` methods of K256.
```

Comment:
It might be a bit more straightforward to use `RecoverableSignPrimitive:: try_sign_recoverable_prehashed` (which is impl'd on `k256::Scalar`) for this:

https://docs.rs/ecdsa/0.8.3/ecdsa/hazmat/trait.RecoverableSignPrimitive.html

The API to compute an ephemeral scalar with RFC6979 is public:

https://docs.rs/ecdsa/0.8.3/ecdsa/rfc6979/fn.generate_k.html

You can convert the raw message digest to a scalar using `Scalar::from_bytes_reduced`